ports/unix/modffi.c

```diff
@@ -493,10 +529,21 @@ STATIC void fficallback_print(const mp_print_t *print, mp_obj_t self_in, mp_prin
     mp_printf(print, "<fficallback %p>", self->func);
 }
 
+STATIC mp_int_t fficallback_get_buffer(mp_obj_t self_in, mp_buffer_info_t *bufinfo, mp_uint_t flags) {
+    (void)flags;
+    mp_obj_fficallback_t *self = MP_OBJ_TO_PTR(self_in);
+
+    bufinfo->buf = &self->func;
```

Comment:
This doesn't feel like the right approach, using the buffer protocol to get out the C closure wrapper function (I would expect it to get access to the underlying data of the function itself).

Instead I'd suggest to add a method to the callback type, eg `cfun()`.  Then one can write:
```python
sa.sa_handler = cb.cfun()
```

Reply:
Done.  

However, `sa.sa_handler = cb.cfun()` doesn't work in this case because we want to set the pointer value, not the value the pointer points at.   

So we need something like this instead:
```python
    memoryview(sa.sa_handler)[:] = cb.cfun()
```

Reply:
> However, `sa.sa_handler = cb.cfun()` doesn't work in this case because we want to set the pointer value, not the value the pointer points at.

If `sa_handler` were defined as a UINT64 would it work without the memoryview?

Reply:
Yes, you are right.
I changed `cfun` to return ULL integer and defined `sa.sa_handler` as `uctypes.UINT64`, it works fine.